### ISSUE TITLE
Individual frontend redirect throwing nil pointer

### ISSUE THREAD

**Issue by gavinbunney** [containous/traefik#2540]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

I'm testing out v1.5-rc2 with the changes from #2133 (great btw!), but traefik is throwing a go nil pointer during start up.

My setup is using a docker backend for user-facing web content (which needs to be on HTTPS) and another consul backend for HTTP or HTTPS traffic. The support from #2133 means I can now run the consul backend services on HTTP whilst leaving the docker backend services on forced HTTPS.

### What did you expect to see?

Traefik starting up without issues, with HTTPS redirect on for the docker backend, and no redirect for Consul backend.

### What did you see instead?

Stack in logs:

```shell
time="2017-12-07T16:41:37Z" level=debug msg="Configuration received from provider docker: {"backends":{"backend-api":{"servers":{"server-server_batch_1":{"url":"http://172.18.0.2:5001","weight":0}},"loadBalancer":{"method":"wrr"}},"backend-web":{"servers":{"server-server_web_1":{"url":"http://172.18.0.3:5000","weight":0}},"loadBalancer":{"method":"wrr"}}},"frontends":{"frontend-PathPrefix-0":{"entryPoints":["http","https"],"backend":"backend-web","routes":{"route-frontend-PathPrefix-0":{"rule":"PathPrefix:/"}},"passHostHeader":true,"priority":0,"basicAuth":[],"headers":{},"redirect":"https"},"frontend-PathPrefixStrip-batch-1":{"entryPoints":["http","https"],"backend":"backend-api","routes":{"route-frontend-PathPrefixStrip-batch-1":{"rule":"PathPrefixStrip:/batch"}},"passHostHeader":true,"priority":0,"basicAuth":[],"headers":{},"redirect":"https"}}}" 
time="2017-12-07T16:41:37Z" level=debug msg="Creating frontend frontend-PathPrefix-0" 
time="2017-12-07T16:41:37Z" level=debug msg="Wiring frontend frontend-PathPrefix-0 to entryPoint http" 
time="2017-12-07T16:41:37Z" level=debug msg="Creating route route-frontend-PathPrefix-0 PathPrefix:/" 
time="2017-12-07T16:41:37Z" level=debug msg="Creating backend backend-web" 
time="2017-12-07T16:41:37Z" level=debug msg="Creating load-balancer wrr" 
time="2017-12-07T16:41:37Z" level=debug msg="Creating server server-server_web_1 at http://172.18.0.3:5000 with weight 0" 
time="2017-12-07T16:41:37Z" level=error msg="Error in Go routine: runtime error: invalid memory address or nil pointer dereference" 
goroutine 13 [running]:
runtime/debug.Stack(0x74e8bf, 0xc4200842d0, 0x211c8b0)
	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
runtime/debug.PrintStack()
	/usr/local/go/src/runtime/debug/stack.go:16 +0x22
github.com/containous/traefik/safe.defaultRecoverGoroutine(0x1dce1e0, 0x32d46c0)
	/go/src/github.com/containous/traefik/safe/routine.go:148 +0x7d
github.com/containous/traefik/safe.GoWithRecover.func1.1(0x219eab8)
	/go/src/github.com/containous/traefik/safe/routine.go:139 +0x57
panic(0x1dce1e0, 0x32d46c0)
	/usr/local/go/src/runtime/panic.go:491 +0x283
github.com/containous/traefik/server.(*Server).buildRedirect(0xc420738000, 0x2100642, 0x5, 0xc420284380, 0xc4203c15b0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/containous/traefik/server/server.go:1314 +0x4a
github.com/containous/traefik/server.(*Server).loadConfig(0xc420738000, 0xc4208d38c0, 0xc42061f340, 0x0, 0x100, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/containous/traefik/server/server.go:1136 +0x2196
github.com/containous/traefik/server.(*Server).loadConfiguration(0xc420738000, 0x210194e, 0x6, 0xc4206748d0)
	/go/src/github.com/containous/traefik/server/server.go:434 +0x224
github.com/containous/traefik/server.(*Server).listenConfigurations(0xc420738000, 0xc420040310)
	/go/src/github.com/containous/traefik/server/server.go:418 +0x51
github.com/containous/traefik/server.(*Server).Start.func2(0xc420040310)
	/go/src/github.com/containous/traefik/server/server.go:198 +0x34
github.com/containous/traefik/safe.(*Pool).Go.func1()
	/go/src/github.com/containous/traefik/safe/routine.go:78 +0x3a
github.com/containous/traefik/safe.GoWithRecover.func1(0x219eab8, 0xc4205f3200)
	/go/src/github.com/containous/traefik/safe/routine.go:142 +0x4d
created by github.com/containous/traefik/safe.GoWithRecover
	/go/src/github.com/containous/traefik/safe/routine.go:136 +0x49
```

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.5.0-rc2
Codename:     cancoillotte
Go version:   go1.9.2
Built:        2017-12-06_03:07:42PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
defaultEntryPoints = ["http", "https"]

[entryPoints]
  [entryPoints.http]
  address = ":80"
  [entryPoints.https]
  address = ":443"
    [entryPoints.https.tls]
      [[entryPoints.https.tls.certificates]]
        CertFile = "server.crt"
        KeyFile = "server.key"

# Enable Docker configuration backend
[docker]
endpoint = "unix:///var/run/docker.sock"
domain = "guidewire.com"
watch = true
exposedbydefault = false

# Enable Consul Catalog configuration backend
[consulCatalog]
endpoint = "fleet-staging:8500"
domain = "deployment.fleet-staging.guidewire.com"
exposedByDefault = true
prefix = "traefik"

```

*docker-compose.yml*

```yml
version: '2'

services:
  proxy:
    image: dtr.guidewire.com/devops/traefik:1.5.0-rc2-guidewire-ssl
    command: --docker --docker.domain=docker.localhost --logLevel=DEBUG
    restart: always
    networks:
      - webgateway
    ports:
      - "80:80"
      - "443:443"
      - "8080:8080"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - /opt/fleet/server/traefik.toml:/traefik.toml

  batch:
    image: harbor.guidewire.com/fleet/fleet-server:0.2.5-batch
    restart: always
    networks:
      - webgateway
    labels:
      - "traefik.backend=api"
      - "traefik.frontend.rule=PathPrefixStrip:/batch"
      - "traefik.frontend.redirect=https"
      - "traefik.port=5001"
    environment:
      - SPRING_PROFILES_ACTIVE=staging

  web:
    image: harbor.guidewire.com/fleet/fleet-server:0.2.5-web
    restart: always
    networks:
      - webgateway
    labels:
      - "traefik.backend=web"
      - "traefik.frontend.rule=PathPrefix:/"
      - "traefik.frontend.redirect=https"
      - "traefik.port=5000"
    environment:
      - SPRING_PROFILES_ACTIVE=staging

networks:
  webgateway:
    driver: bridge
```

### If applicable, please paste the log output in debug mode (`--debug` switch)

```shell
time="2017-12-07T16:41:37Z" level=debug msg="Global configuration loaded {"LifeCycle":{"RequestAcceptGraceTimeout":0,"GraceTimeOut":0},"GraceTimeOut":0,"Debug":false,"CheckNewVersion":true,"SendAnonymousUsage":false,"AccessLogsFile":"","AccessLog":null,"TraefikLogsFile":"","TraefikLog":null,"LogLevel":"DEBUG","EntryPoints":{"http":{"Network":"","Address":":80","TLS":null,"Redirect":null,"Auth":null,"WhitelistSourceRange":null,"Compress":false,"ProxyProtocol":null,"ForwardedHeaders":{"Insecure":true,"TrustedIPs":null}},"https":{"Network":"","Address":":443","TLS":{"MinVersion":"","CipherSuites":null,"Certificates":[{"CertFile":"server.crt","KeyFile":"server.key"}],"ClientCAFiles":null,"ClientCA":{"Files":null,"Optional":false}},"Redirect":null,"Auth":null,"WhitelistSourceRange":null,"Compress":false,"ProxyProtocol":null,"ForwardedHeaders":{"Insecure":true,"TrustedIPs":null}}},"Cluster":null,"Constraints":[],"ACME":null,"DefaultEntryPoints":["http","https"],"ProvidersThrottleDuration":2000000000,"MaxIdleConnsPerHost":200,"IdleTimeout":0,"InsecureSkipVerify":false,"RootCAs":null,"Retry":null,"HealthCheck":{"Interval":30000000000},"RespondingTimeouts":null,"ForwardingTimeouts":null,"Web":null,"Docker":{"Watch":true,"Filename":"","Constraints":null,"Trace":false,"DebugLogGeneratedTemplate":false,"Endpoint":"unix:///var/run/docker.sock","Domain":"docker.localhost","TLS":null,"ExposedByDefault":true,"UseBindPortIP":false,"SwarmMode":false},"File":null,"Marathon":null,"Consul":null,"ConsulCatalog":{"Watch":false,"Filename":"","Constraints":[],"Trace":false,"DebugLogGeneratedTemplate":false,"Endpoint":"fleet-staging:8500","Domain":"deployment.fleet-staging.guidewire.com","ExposedByDefault":true,"Prefix":"traefik","FrontEndRule":"Host:{{.ServiceName}}.{{.Domain}}"},"Etcd":null,"Zookeeper":null,"Boltdb":null,"Kubernetes":null,"Mesos":null,"Eureka":null,"ECS":null,"Rancher":null,"DynamoDB":null,"ServiceFabric":null,"Rest":null,"API":null,"Metrics":null,"Ping":null}" 
time="2017-12-07T16:41:37Z" level=info msg="Preparing server http &{Network: Address::80 TLS:<nil> Redirect:<nil> Auth:<nil> WhitelistSourceRange:[] Compress:false ProxyProtocol:<nil> ForwardedHeaders:0xc420625980} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s" 
time="2017-12-07T16:41:37Z" level=info msg="Preparing server https &{Network: Address::443 TLS:0xc420732d80 Redirect:<nil> Auth:<nil> WhitelistSourceRange:[] Compress:false ProxyProtocol:<nil> ForwardedHeaders:0xc420625960} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s" 
time="2017-12-07T16:41:37Z" level=debug msg="Add certificate for domains *.guidewire.com*.guidewire.com,*.guidewire.com,guidewire.com" 
time="2017-12-07T16:41:37Z" level=info msg="Starting provider *docker.Provider {"Watch":true,"Filename":"","Constraints":null,"Trace":false,"DebugLogGeneratedTemplate":false,"Endpoint":"unix:///var/run/docker.sock","Domain":"docker.localhost","TLS":null,"ExposedByDefault":true,"UseBindPortIP":false,"SwarmMode":false}" 
time="2017-12-07T16:41:37Z" level=info msg="Starting provider *consul.CatalogProvider {"Watch":false,"Filename":"","Constraints":[],"Trace":false,"DebugLogGeneratedTemplate":false,"Endpoint":"fleet-staging:8500","Domain":"deployment.fleet-staging.guidewire.com","ExposedByDefault":true,"Prefix":"traefik","FrontEndRule":"Host:{{.ServiceName}}.{{.Domain}}"}" 
time="2017-12-07T16:41:37Z" level=info msg="Starting server on :80" 
time="2017-12-07T16:41:37Z" level=info msg="Starting server on :443" 
time="2017-12-07T16:41:37Z" level=debug msg="Provider connection established with docker 17.06.2-ce (API 1.30)" 
time="2017-12-07T16:41:37Z" level=debug msg="Could not load traefik.frontend.whitelistSourceRange labels" 
time="2017-12-07T16:41:37Z" level=debug msg="Could not load traefik.frontend.entryPoints labels" 
time="2017-12-07T16:41:37Z" level=debug msg="Could not load traefik.frontend.auth.basic labels" 
time="2017-12-07T16:41:37Z" level=debug msg="Could not load traefik.frontend.whitelistSourceRange labels" 
time="2017-12-07T16:41:37Z" level=debug msg="Could not load traefik.frontend.entryPoints labels" 
time="2017-12-07T16:41:37Z" level=debug msg="Could not load traefik.frontend.auth.basic labels" 
time="2017-12-07T16:41:37Z" level=debug msg="Fetching service" service=service-devops-hello-fleet-backend 
time="2017-12-07T16:41:37Z" level=debug msg="Validation of load balancer method for backend backend-api failed: invalid load-balancing method ''. Using default method wrr." 
time="2017-12-07T16:41:37Z" level=debug msg="Validation of load balancer method for backend backend-web failed: invalid load-balancing method ''. Using default method wrr." 
time="2017-12-07T16:41:37Z" level=debug msg="Configuration received from provider docker: {"backends":{"backend-api":{"servers":{"server-server_batch_1":{"url":"http://172.18.0.2:5001","weight":0}},"loadBalancer":{"method":"wrr"}},"backend-web":{"servers":{"server-server_web_1":{"url":"http://172.18.0.3:5000","weight":0}},"loadBalancer":{"method":"wrr"}}},"frontends":{"frontend-PathPrefix-0":{"entryPoints":["http","https"],"backend":"backend-web","routes":{"route-frontend-PathPrefix-0":{"rule":"PathPrefix:/"}},"passHostHeader":true,"priority":0,"basicAuth":[],"headers":{},"redirect":"https"},"frontend-PathPrefixStrip-batch-1":{"entryPoints":["http","https"],"backend":"backend-api","routes":{"route-frontend-PathPrefixStrip-batch-1":{"rule":"PathPrefixStrip:/batch"}},"passHostHeader":true,"priority":0,"basicAuth":[],"headers":{},"redirect":"https"}}}" 
time="2017-12-07T16:41:37Z" level=debug msg="Creating frontend frontend-PathPrefix-0" 
time="2017-12-07T16:41:37Z" level=debug msg="Wiring frontend frontend-PathPrefix-0 to entryPoint http" 
time="2017-12-07T16:41:37Z" level=debug msg="Creating route route-frontend-PathPrefix-0 PathPrefix:/" 
time="2017-12-07T16:41:37Z" level=debug msg="Creating backend backend-web" 
time="2017-12-07T16:41:37Z" level=debug msg="Creating load-balancer wrr" 
time="2017-12-07T16:41:37Z" level=debug msg="Creating server server-server_web_1 at http://172.18.0.3:5000 with weight 0" 
time="2017-12-07T16:41:37Z" level=error msg="Error in Go routine: runtime error: invalid memory address or nil pointer dereference" 
goroutine 13 [running]:
runtime/debug.Stack(0x74e8bf, 0xc4200842d0, 0x211c8b0)
	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
runtime/debug.PrintStack()
	/usr/local/go/src/runtime/debug/stack.go:16 +0x22
github.com/containous/traefik/safe.defaultRecoverGoroutine(0x1dce1e0, 0x32d46c0)
	/go/src/github.com/containous/traefik/safe/routine.go:148 +0x7d
github.com/containous/traefik/safe.GoWithRecover.func1.1(0x219eab8)
	/go/src/github.com/containous/traefik/safe/routine.go:139 +0x57
panic(0x1dce1e0, 0x32d46c0)
	/usr/local/go/src/runtime/panic.go:491 +0x283
github.com/containous/traefik/server.(*Server).buildRedirect(0xc420738000, 0x2100642, 0x5, 0xc420284380, 0xc4203c15b0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/containous/traefik/server/server.go:1314 +0x4a
github.com/containous/traefik/server.(*Server).loadConfig(0xc420738000, 0xc4208d38c0, 0xc42061f340, 0x0, 0x100, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/containous/traefik/server/server.go:1136 +0x2196
github.com/containous/traefik/server.(*Server).loadConfiguration(0xc420738000, 0x210194e, 0x6, 0xc4206748d0)
	/go/src/github.com/containous/traefik/server/server.go:434 +0x224
github.com/containous/traefik/server.(*Server).listenConfigurations(0xc420738000, 0xc420040310)
	/go/src/github.com/containous/traefik/server/server.go:418 +0x51
github.com/containous/traefik/server.(*Server).Start.func2(0xc420040310)
	/go/src/github.com/containous/traefik/server/server.go:198 +0x34
github.com/containous/traefik/safe.(*Pool).Go.func1()
	/go/src/github.com/containous/traefik/safe/routine.go:78 +0x3a
github.com/containous/traefik/safe.GoWithRecover.func1(0x219eab8, 0xc4205f3200)
	/go/src/github.com/containous/traefik/safe/routine.go:142 +0x4d
created by github.com/containous/traefik/safe.GoWithRecover
	/go/src/github.com/containous/traefik/safe/routine.go:136 +0x49
```

**Comment by traefiker** [comment]
Closed by #2544.

**Comment by gavinbunney** [comment]
Excellent thanks!
